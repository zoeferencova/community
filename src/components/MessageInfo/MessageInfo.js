import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import { PropTypes } from 'prop-types';
import CommUnityContext from "../../contexts/context";
import { ButtonLight, ButtonDark } from "../Utils/Utils";
import styles from "./MessageInfo.module.css";

class MessageInfo extends Component {
    static contextType = CommUnityContext;

    // Redirects user to the post that originated the chat
    handleGoToPost(e, post) {
        e.preventDefault();
        if (post.user_id === this.context.user.id) {
            this.props.history.push(`/my-post/${post.id}`)
        } else {
            this.props.history.push(`/post/${post.id}`)
        }
        
    }

    // Formats information sentence based on type of post and whether the post was written by the current user or the other user in the chat
    formatSentence(post, user) {
        if (post.user_id === this.context.user.id) {
            if (post.post_type === "offer") {
                return `${user.first_name} offered to help you`
            } else {
                return `You accepted help from ${user.first_name}`
            }
        } else if (post.user_id === user.id) {
            if (post.post_type === "offer") {
                return `${user.first_name} offered to help you`
            } else {
                return `You offered to help ${user.first_name}`
            }
        }
    }

    render() {
        const { post } = this.context.activeChat;
        const { user } = this.props;
        return (
            <div className={styles.container}>
                <p>{this.formatSentence(post, user)}<span>{post.post_type === "offer" ? <i className={`fas fa-heart ${styles.heart}`}></i> : <i className={`fas fa-hand-paper ${styles.hand}`}></i>}</span></p>
                <div className={styles.buttons}>
                    <ButtonDark type="button" onClick={e => this.handleGoToPost(e, post)} id="post-button">Go to post</ButtonDark>
                    <ButtonLight className={styles.deleteButton} type="button" onClick={e => this.props.deleteChat(e)} loading={this.props.loading.toString()} id="delete-button">Delete chat</ButtonLight>
                </div>
            </div>
        );
    }
}

export default withRouter(MessageInfo);

MessageInfo.propTypes = {
    user: PropTypes.object,
    deleteChat: PropTypes.func,
    loading: PropTypes.bool
}