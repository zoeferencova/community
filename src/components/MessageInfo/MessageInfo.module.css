.container {
    height: 90px;
    position: absolute;
    top: 0;
    z-index: 1000;
    top: 75px;
    background-color: #f2f5f7;
    display: flex;
    padding: 0px 15px;
    left: 10px; 
    right: 10px;
    border-radius: 10px;
    align-items: center;
    justify-content: space-between;
    animation: fadeInDown ease 0.3s;
    -webkit-animation: fadeInDown ease 0.3s;
    -moz-animation: fadeInDown ease 0.3s;
    -o-animation: fadeInDown ease 0.3s;
    -ms-animation: fadeInDown ease 0.3s;
}

.container p {
   color: black;
   font-size: 15px;
   margin-left: 15px;
   text-align: left;
}

.buttons {
   display: flex;
   flex-direction: row;
}


.buttons button {
   margin: 0px 10px;
}

.deleteButton {
   color: #ad3434;
   border: 1px solid #ad3434;
}

.deleteButton:hover {
   border: 1px solid #b55151;
}

.heart, .hand {
   margin-left: 6px;
   font-size: 16px;
}

.heart {
   color: #bf5e6b;
}

.hand {
   color: #e6c78e;
}

@keyframes fadeInDown {
    0% {
       opacity: 0;
       transform: translateY(-10px);
    }
    100% {
       opacity: 1;
       transform: translateY(0);
    }
} 

@-moz-keyframes fadeInDown {
    0% {
       opacity: 0;
       transform: translateY(-10px);
    }
    100% {
       opacity: 1;
       transform: translateY(0);
    }
} 

@-webkit-keyframes fadeInDown {
    0% {
       opacity: 0;
       transform: translateY(-10px);
    }
    100% {
       opacity: 1;
       transform: translateY(0);
    }
} 

@-o-keyframes fadeInDown {
    0% {
       opacity: 0;
       transform: translateY(-10px);
    }
    100% {
       opacity: 1;
       transform: translateY(0);
    }
} 

@-ms-keyframes fadeInDown {
    0% {
       opacity: 0;
       transform: translateY(-10px);
    }
    100% {
       opacity: 1;
       transform: translateY(0);
    }
} 

@media only screen and (max-width: 608px) {
   .container {
      flex-wrap: wrap;
      height: 130px;
      justify-content: center;
   }

   .container .buttons {
      margin-bottom: 20px;
   }

}