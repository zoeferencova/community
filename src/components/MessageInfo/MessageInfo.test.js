import React from "react";
import { render } from "enzyme";
import MessageInfo from "./MessageInfo";
import { BrowserRouter } from "react-router-dom";
import CommUnityContext from "../../contexts/context";
import mockContext from "../../contexts/mock-context";

describe("MessageInfo", () => {
    it("renders MessageInfo component given mock context data", () => {
        const mockProps = {
            user: {
                email: "james@gmail.com",
                first_name: "James",
                id: 2,
                location: {lat: 40.748225, lng: -73.89698620000001},
                radius: "3.00"
            },
            loading: false
        }

        const wrapper = render(
            <CommUnityContext.Provider value={mockContext}>
                <BrowserRouter>
                    <MessageInfo {...mockProps} />
                </BrowserRouter>
            </CommUnityContext.Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});