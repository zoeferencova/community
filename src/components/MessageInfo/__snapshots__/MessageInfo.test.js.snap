// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInfo renders MessageInfo component given mock context data 1`] = `
<div
  class="container"
>
  <p>
    James offered to help you
    <span>
      <i
        class="fas fa-heart heart"
      />
    </span>
  </p>
  <div
    class="buttons"
  >
    <button
      class="buttonDark "
      type="button"
    >
      Go to post
    </button>
    <button
      class="button deleteButton"
      loading="false"
      type="button"
    >
      Delete chat
    </button>
  </div>
</div>
`;
